orchagent/acltable.cpp

```diff
@@ -0,0 +1,436 @@
+#include "logger.h"
+
+#include "port.h"
+#include "acltable.h"
+#include "aclorch.h"
+#include "crmorch.h"
+
+using namespace std;
+
+extern sai_acl_api_t*    sai_acl_api;
+extern sai_object_id_t   gSwitchId;
+extern PortsOrch*        gPortsOrch;
+extern CrmOrch*          gCrmOrch;
+
+bool AclTable::validate()
+{
+    if (type == ACL_TABLE_CTRLPLANE)
+        return true;
+
+    if (type == ACL_TABLE_UNKNOWN || stage == ACL_STAGE_UNKNOWN)
+        return false;
+
+    if (portSet.empty() && pendingPortSet.empty())
+        return false;
+
+    return true;
+}
+
+bool AclTable::create()
```

Comment:
Just a note, this method is kinda large and there are a lot of conditional checks for different types of tables and it gets a bit confusing in places. I think it would be worth breaking this up into private helpers to handle different table types, but that is also a pretty big change and probably out of the scope of this PR. Can you leave a note about it?